provider/resource.go

```diff
@@ -235,6 +237,60 @@ func detectCloudShell() cloudShellProfile {
 	return negative
 }
 
+// stringValue gets a string value from a property map if present, else ""
+func stringValue(vars resource.PropertyMap, prop resource.PropertyKey, envs []string) string {
+	val, ok := vars[prop]
+	if ok && val.IsString() {
+		return val.StringValue()
+	}
+	for _, env := range envs {
+		val, ok := os.LookupEnv(env)
+		if ok {
+			return val
+		}
+	}
+	return ""
+}
+
+// preConfigureCallback returns an error when cloud provider setup is misconfigured
+func preConfigureCallback(vars resource.PropertyMap, c tfshim.ResourceConfig) error {
+
+	envName := stringValue(vars, "environment", []string{"ARM_ENVIRONMENT"})
+	if envName == "" {
+		envName = "public"
+	}
+
+	// validate the azure config
+	// make a Builder
+	builder := &authentication.Builder{
```

Comment:
Ideally, we would find a way to share this builder code with `provider.go:getAuthConfig`. I assume the main issue is that we don't have the provider config yet for `k.getConfig`? We could abstract the lookup of a single property into a `configGetter` interface and pass it in, for instance.

Reply:
I'm not sure I understand your comment entirely - we currently [do not have CheckConfig enabled](https://github.com/pulumi/pulumi-terraform-bridge/blob/master/pkg/tfbridge/provider.go#L314) in the TF bridge, and would have to implement such an interface there so we can capture an auth config getter for any cloud? 
here's the issue: https://github.com/pulumi/pulumi-terraform-bridge/issues/244

Reply:
Oops, I was confused and was talking about provider.go in the native azure provider. I has a very similar func, but I guess sharing code is not practical.